gene_splicer/utils.py

```diff
@@ -390,15 +391,67 @@ def align(target_seq,
     else:
         return alignment_path
 
+HIVINTACT_ERRORS_TABLE = [
+    'AlignmentFailed',
+    'InvalidCodon',
+    'NonHIV',
+    'LongDeletion',
+    'InternalInversion',
+    'Scramble',
+    'APOBECHypermutationDetected',
+    'MajorSpliceDonorSiteMutated',
+    'PackagingSignalDeletion',
+    'PackagingSignalNotComplete',
+    'RevResponseElementDeletion',
+    'MisplacedORF',
+    'WrongORFNumber',
+    'DeletionInOrf',
+    'InsertionInOrf',
+    'InternalStopInOrf',
+    'FrameshiftInOrf',
+    ]
+
+def iterate_hivintact_data(name, outpath):
+    intact = {}
+
+    def get_verdict(SEQID, all_errors):
+        ordered = sorted(all_errors, key=HIVINTACT_ERRORS_TABLE.index)
+        verdict = ordered[0]
+        return [SEQID, verdict]
+
+    for d in glob.glob(str(outpath / 'hivintact*')):
+        for (SEQID, sequence) in read_fasta(os.path.join(d, 'intact.fasta')):
+            row = [SEQID, 'Intact']
+            intact[SEQID] = True
+            yield row
 
-def generate_table_precursor(name, outpath, add_columns=None):
-    # Output csv
-    precursor_path: Path = outpath / 'table_precursor.csv'
+        sequence_name = None
+        with open(os.path.join(d, 'errors.csv'), 'r') as f:
+            reader = csv.DictReader(f)
 
-    # Load filtered sequences
-    filtered_path = outpath / (name + '_filtered.csv')
-    filtered = pd.read_csv(filtered_path)
-    # Load hivseqinr data
+            last_name = None
+            all_errors = []
+            for row in reader:
```

Comment:
Consider using `groupby()` to simplify this code.

Reply:
Thank you, it does simplify the code quite a lot!